C error when dynamic casting a struct field to none

I've only (re)started learning V today so I'm not sure if the issue is that I'm able to have a sum type with none at all, or just using the 'is' operator with none.

It works properly if I use an empty struct as a fake none instead.

Also as v doctor shows, there's no cc on my PATH, but I don't think that's related since V builds properly with other code.

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: windows, Microsoft Windows 11 Home v22000 64-bit
Processor: 24 cpus, 64bit, little endian, 12th Gen Intel(R) Core(TM) i9-12900KF  

CC version: Error: exec failed (CreateProcess) with code 2: The system cannot find the file specified.

 cmd: cc --version

getwd: C:\Users\misabiko\Documents\Coding\Projects\V\TheGreatFollowing - Copy
vmodules: C:\Users\misabiko\.vmodules
vroot: C:\Users\misabiko\Documents\Coding\V\v
vexe: C:\Users\misabiko\Documents\Coding\V\v\v.exe
vexe mtime: 2022-08-20 22:58:10
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 c356882.eb6627f

Git version: git version 2.37.2.windows.2
Git vroot status: weekly.2022.33-38-geb6627fe
.git/config present: true
thirdparty/tcc status: thirdparty-windows-amd64 125e21e2

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg .\none_field_check.v`
```v
module main

struct None {}

struct MyStruct {
	my_first_field string | none
	my_second_field string | none
	my_third_field string | None
}

fn main() {
	s := MyStruct{
		my_first_field: 'blah'
		my_second_field: none
		my_third_field: None{}
	}

	//Works
	println(if s.my_first_field is string {
		'first field is none'
	} else {
		'first field is a string'
	})

	//Error comes up when casting to none
	println(if s.my_second_field is none {
		'second field is none'
	} else {
		'second field is a string'
	})

	//Works
	println(if s.my_third_field is None {
		'third field is None'
	} else {
		'third field is a string'
	})
}

```

**What did you expect to see?**

"s.my_second_field is none" should evaluate to true and print accordingly, or there should be a build error indicating why this doesn't work


**What did you see instead?**
```
==================
C:/Users/misabiko/AppData/Local/Temp/v_0/none_field_check.10255314589253006336.tmp.c:7468: warning: cast between pointer and integer of different size
C:/Users/misabiko/AppData/Local/Temp/v_0/none_field_check.10255314589253006336.tmp.c:12360: error: invalid operand types for binary operation
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/infix.v
+++ b/vlib/v/gen/c/infix.v
@@ -619,7 +619,11 @@ fn (mut g Gen) infix_expr_is_op(node ast.InfixExpr) {
 	} else if left_sym.kind == .sum_type {
 		g.write('_typ $cmp_op ')
 	}
-	g.expr(node.right)
+	if node.right is ast.None {
+		g.write('$ast.none_type.idx() /* none */')
+	} else {
+		g.expr(node.right)
+	}
 }
 
 fn (mut g Gen) gen_interface_is_op(node ast.InfixExpr) {

--- a/vlib/v/tests/inout/printing_sumtype_with_none.out
+++ b/vlib/v/tests/inout/printing_sumtype_with_none.out
@@ -1,0 +1,3 @@
+first field is a string
+second field is none
+third field is None

--- a/vlib/v/tests/inout/printing_sumtype_with_none.vv
+++ b/vlib/v/tests/inout/printing_sumtype_with_none.vv
@@ -1,0 +1,35 @@
+module main
+
+struct None {}
+
+struct MyStruct {
+	my_first_field  string|none
+	my_second_field string|none
+	my_third_field  string|None
+}
+
+fn main() {
+	s := MyStruct{
+		my_first_field: 'blah'
+		my_second_field: none
+		my_third_field: None{}
+	}
+
+	println(if s.my_first_field is string {
+		'first field is a string'
+	} else {
+		'first field is none'
+	})
+
+	println(if s.my_second_field is none {
+		'second field is none'
+	} else {
+		'second field is a string'
+	})
+
+	println(if s.my_third_field is None {
+		'third field is None'
+	} else {
+		'third field is a string'
+	})
+}

